routing/routing_integration_tests/routing_test_tools.cpp

```diff
@@ -81,14 +83,15 @@ namespace integration
     return osrmRouter;
   }
 
-  shared_ptr<IRouter> CreatePedestrianRouter(Index & index,
-                                             storage::CountryInfoGetter const & infoGetter)
+  shared_ptr<IRouter> CreateAStarRouter(Index & index,
+                                        storage::CountryInfoGetter const & infoGetter,
+                                        TRouterFactory const & routerFactory)
   {
     auto countryFileGetter = [&infoGetter](m2::PointD const & pt)
     {
       return infoGetter.GetRegionCountryId(pt);
     };
-    unique_ptr<IRouter> router = CreatePedestrianAStarBidirectionalRouter(index, countryFileGetter);
+    unique_ptr<IRouter> router = routerFactory(index, countryFileGetter);
```

Comment:
infoGetter? It's a member of IRouterComponents::m_infoGetter. IRouterComponents(OsrmRouterComponents/PedestrianRouterComponents/BicycleRouterComponents). This component is created in the beginning of an integration test, is existed during the integration test and is destroyed after the test is finished.